### PR TITLE
Refs #22513 - Allow overriding postgresql server settings via hiera

### PR DESCRIPTION
**Pull request by chris1984**

### CODE REVIEW

File: config/foreman.hiera/custom.yaml [hunk 1]
```diff
@@ -24,3 +24,7 @@
 # apache::mod::prefork::serverlimit: 256
 # apache::mod::prefork::maxclients: 256
 # apache::mod::prefork::maxrequestsperchild: 4000
+# Here are some examples of how you tune the Postgres options if needed:
```
Comment (ekohl) [inline config/foreman.hiera/custom.yaml]: I'd add an empty line above this rather than below it.

File: config/foreman.hiera/custom.yaml [hunk 2]
```diff
@@ -24,3 +24,7 @@
...
+#
+# postgresql::server::max_connections: 200
```
Comment (ekohl) [inline config/foreman.hiera/custom.yaml]: It should be a hash:
```yaml
postgresql::server::config_entries:
  max_connections: 200
  shared_buffers: 256MB
```

**Review by ekohl — CHANGES_REQUESTED** [review]

**Comment by chris1984** [comment]
@ekohl updated added an empty line above and below so it is not all jammed together and easy to read.

**Review by ekohl — APPROVED** [review]
This is pending a puppetlabs-postgresql release, but other than that I think this is correct.

**Comment by ehelms** [comment]
@ekohl If possible, I'd like to get this in before we branch the installer, which would include possibly setting postgres to nightly (or at least the commit we are wanting to include). I think you said in the past that would be OK, but wondering when you are seeing the branch point of the installer is to see if we could go ahead and do this.

File: Puppetfile [hunk 3]
```diff
@@ -2,7 +2,7 @@ forge 'https://forgeapi.puppetlabs.com'
 
 # Dependencies
 mod 'puppetlabs/mysql',         '>= 4.0.0'
-mod 'puppetlabs/postgresql',    '>= 4.8.0'
+mod 'puppetlabs/postgresql',    :git => 'https://github.com/puppetlabs/puppetlabs-postgresql'
```
Comment (ekohl) [inline Puppetfile]: Could you add a comment linking to https://tickets.puppetlabs.com/browse/MODULES-6963 so we know why it's there?

**Review by ekohl — CHANGES_REQUESTED** [review]
I think the commit message could be clearer. How about `Allow overriding postgresql server settings via hiera`

**Comment by chris1984** [comment]
@ekohl updated

File: config/foreman.hiera/custom.yaml [hunk 4]
```diff
@@ -24,3 +24,9 @@
 # apache::mod::prefork::serverlimit: 256
 # apache::mod::prefork::maxclients: 256
 # apache::mod::prefork::maxrequestsperchild: 4000
+#
+# Here are some examples of how you tune the Postgres options if needed:
```
Comment (ekohl) [inline config/foreman.hiera/custom.yaml]: I think the official spelling is PostgreSQL. @evgeni has been persistent in pointing this out so let's keep it correct.

**Comment by chris1984** [comment]
@ekohl updated

**Review by ekohl — APPROVED** [review]
@mmoll what do you think?

**Review by mmoll — CHANGES_REQUESTED** [review]
I'd rather fork that repo and have a defined state and not tap a foreign master branch

**Comment by ehelms** [comment]
@mmoll Would setting Puppetfile to point at the commit in master we minimally want suffice for the time being?

**Comment by mmoll** [comment]
@ehelms After having a look at the changes over there: Yes, that sounds sensible. :+1:

**Comment by chris1984** [comment]
@mmoll how does this look?

File: Puppetfile [hunk 5]
```diff
@@ -2,8 +2,8 @@ forge 'https://forgeapi.puppetlabs.com'
 
 # Dependencies
 mod 'puppetlabs/mysql',         '>= 4.0.0'
-mod 'puppetlabs/postgresql',    '>= 4.8.0'
-mod 'puppetlabs/puppetdb'
+# https://tickets.puppetlabs.com/browse/MODULES-6963
+mod 'puppetlabs/postgresql',    :git => 'https://github.com/puppetlabs/puppetlabs-postgresql' , :ref => 'Parameters `roles`, `config_entires`, and `pg_hba_rules` to `postgresql::server` for hiera'
```
Comment (mmoll) [inline Puppetfile]: please use the git commit id here

Comment (ekohl) [reply Puppetfile]: I'd also prefer the current latest commit in master since there are docs changes as well as others.

**Review by ekohl — CHANGES_REQUESTED** [review]

**Comment by chris1984** [comment]
@ekohl @mmoll how does it look now?

**Review by mmoll — DISMISSED** [review]

File: Puppetfile [hunk 6]
```diff
@@ -2,8 +2,7 @@ forge 'https://forgeapi.puppetlabs.com'
 
 # Dependencies
 mod 'puppetlabs/mysql',         '>= 4.0.0'
-mod 'puppetlabs/postgresql',    '>= 4.8.0'
-mod 'puppetlabs/puppetdb'
```
Comment (ekohl) [inline Puppetfile]: Any reason you're removing puppetdb?

**Review by mmoll — APPROVED** [review]

**Comment by chris1984** [comment]
Updated

**Review by ekohl — APPROVED** [review]

**Comment by mmoll** [comment]
merged, thanks @chris1984!

**Pull request merged**